src/lib.rs

```diff
@@ -702,6 +702,31 @@ where
         }
     }
 
+    /// Returns an iterator that visits all values present in `self` that is not
+    /// present in `other`. Takes the number of occurrences into account in both
+    /// bags.
```

Comment:
This comment sent me down a rabbit hole of trying to figure out what the semantics of "difference" for multisets even should be. [This StackOverflow post](https://math.stackexchange.com/questions/2659699/set-theoretic-difference-of-multisets) suggests that given `A = { 1, 1, 2 }` and `B = { 1 }`, the difference (`A.difference(B)`) can feasibly be both `{ 1, 2 }`, and `{ 2 }`. Since that means that the word "difference" here is ambiguous, I think we should instead have two methods: `subtract` and `not_in`. Hopefully it's clear which is expected to have which outcome, otherwise these names are bad too (but I believe this would be `subtract`). What do you think? Happy to have this PR just include one of them if you don't want to implement the other.

Once we've chosen the semantics this function should implement, I think we should also update the docs here to specifically mention the interaction with counts.

Reply:
I'm totally fine with renaming the method to `subtract`. I will do that. I'd prefer to leave `not_in` to be implemented by someone that needs it. Thank you for the understanding.

I'm not sure I completely understand what kind of update the docs you are after. The docs already say "Takes the number of occurrences into account in both bags" which I think alone communicates what the semantics of the method are. That, in combination with the example code, I think makes the semantics very clear? I'm happy to make any changes to the docs you like though, I just need to understand what you are after :)

Reply:
@jonhoo I was watching the stream where this was reviewed, and just wanted to give my two cents on the naming: I think another valid name pair which is more descriptive and direct is `multiplicity_difference` and `membership_difference`, where `multiplicity_difference` corresponds to `subtract` and returns the difference in multiplicities, and `membership_difference` merely takes the difference based on membership, not multiplicity.

Reply:
@Enselic The trouble with "takes the counts into account" is that it doesn't say _how_ it takes the counts into account. For example, does that mean I'll get back negative differences too if something is in B but not A? I'd like us to be quite explicit with something like:

> This method returns an iterator over all the elements that are in `self` with a higher occurrence count than in `other`. The count in the returned iterator represents how many more of a given element are in `self` than `other`.

@Cassy343 I see where you're coming from, though I think both of your proposed names come with challenges of their own:

 - `multiplicity_difference`: This, to me, sounds like something that will return a negative count for something that has a higher multiplicity in `other` than in `self`. Which is not the semantics of this method. The word "multiplicity" is also fairly technical, and would almost certainly make the method harder to understand at a glance.
 - `membership_difference`: I like how accurate this is, but it's also verbose. And I'm not sure it represents a meaningful semantic improvement over `A.not_in(B)`?

Reply:
@jonhoo They are very long and technical, I see your point, and I've seen the discussion about just falling back to plain old `difference` which would be my top choice. However I feel like `not_in` could be confused for `subtract`. If I have a bag with 2 As, and another with 1 A, then there is 1 A in the first bag that is `not_in` the latter. Perhaps something like `none_in` would get the point across more clearly. Also a multiset formally is defined as a pair with a regular set and a function mapping elements of that set to their multiplicities, so calling it `set_difference` would also be pretty mathematically correct (referring to the set part of the pair) and more succinct. Not a big deal since it's not being implemented in this PR but something to think about.

Reply:
I quite like `none_in` actually! But yes, for a different PR.